connector/src/main/java/org/geysermc/connector/network/translators/item/translators/nbt/CrossbowTranslator.java

```diff
@@ -0,0 +1,84 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.connector.network.translators.item.translators.nbt;
+
+import com.github.steveice10.opennbt.tag.builtin.ByteTag;
+import com.github.steveice10.opennbt.tag.builtin.CompoundTag;
+import com.github.steveice10.opennbt.tag.builtin.ListTag;
+import com.github.steveice10.opennbt.tag.builtin.StringTag;
+import org.geysermc.connector.network.translators.ItemRemapper;
+import org.geysermc.connector.network.translators.NbtItemStackTranslator;
+import org.geysermc.connector.network.translators.item.ItemEntry;
+
+@ItemRemapper
+public class CrossbowTranslator extends NbtItemStackTranslator {
+
+    @Override
+    public void translateToBedrock(CompoundTag itemTag, ItemEntry itemEntry) {
+        // This needs sorting to allow for the bedrock client to successfully know the crossbow is loaded
+        // https://github.com/NukkitX/Protocol/pull/41
+        // Unknown EntityEvent 74 in packet EntityEventPacket(runtimeEntityId=7, type=null, data=0)
+        // [CLIENT BOUND]  -  EntityEventPacket(runtimeEntityId=7, type=null, data=0)
+        // [CLIENT BOUND]  -  CompletedUsingItemPacket(itemId=471, action=UNKNOWN)
+
+        if (itemTag.get("ChargedProjectiles") != null) {
+            ListTag chargedProjectiles = itemTag.get("ChargedProjectiles");
+            if (!chargedProjectiles.getValue().isEmpty()) {
+                CompoundTag projectile = (CompoundTag) chargedProjectiles.getValue().get(0);
+
+                CompoundTag newProjectile = new CompoundTag("chargedItem");
+                newProjectile.put(new ByteTag("Count", (byte) projectile.get("Count").getValue()));
+                newProjectile.put(new StringTag("Name", (String) projectile.get("id").getValue()));
+
+                // Not sure what this is for
+                newProjectile.put(new ByteTag("Damage", (byte) 0));
+
+                itemTag.put(newProjectile);
+            }
+        }
+    }
+
+    @Override
+    public void translateToJava(CompoundTag itemTag, ItemEntry itemEntry) {
+        if (itemTag.get("chargedItem") != null) {
+            CompoundTag chargedItem = itemTag.get("chargedItem");
+
+            CompoundTag newProjectile = new CompoundTag("");
+            newProjectile.put(new ByteTag("Count", (byte) chargedItem.get("Count").getValue()));
+            newProjectile.put(new StringTag("id", (String) chargedItem.get("Name").getValue()));
+
+            ListTag chargedProjectiles = new ListTag("ChargedProjectiles");
+            chargedProjectiles.add(newProjectile);
+
+            itemTag.put(chargedProjectiles);
+        }
+    }
+
+    @Override
+    public boolean acceptItem(ItemEntry itemEntry) {
+        return "minecraft:crossbow".equals(itemEntry.getJavaIdentifier());
```

Comment:
Tiny nitpick that isn't required, but I think this would look better the other way.

Reply:
This was done as it prevents errors if the `getJavaIdentifier()` returns null

Reply:
Oh, very smart.